lib/nostrum/struct/channel.ex

```diff
@@ -37,132 +135,282 @@ defmodule Nostrum.Struct.Channel do
+  | `13` | Guild Stage Voice    | `GUILD_STAGE_VOICE`     _A voice channel for hosting events with an audience_          |
+
+  More information about _Discord Channel Types_ can be found on the [Discord API Channel Type Documentation](https://discord.com/developers/docs/resources/channel#channel-object-channel-types).
+  """
+  ## Unsure if it should be here or at the top of the documentation, at both for now.
+  @type type :: integer()
+
+  @typedoc """
+  The id of the guild the channel is located in.
+  """
   @type guild_id :: Guild.id()
 
-  @typedoc "The ordered position of the channel"
+  @typedoc """
+  The position of the channel in the sidebar of the guild.
+  """
   @type position :: integer()
 
-  @typedoc "The list of overwrites"
+  @typedoc """
+  A list of permission overwrites applied to the channel.
+  """
   @type permission_overwrites :: [Overwrite.t()]
 
-  @typedoc "The name of the channel"
+  @typedoc """
+  The name of the channel.
+  """
   @type name :: String.t()
 
-  @typedoc "Current channel topic"
+  @typedoc """
+  The topic of the channel.
+  """
   @type topic :: String.t()
 
-  @typedoc "If the channel is nsfw"
-  @type nsfw :: boolean
+  @typedoc """
+  The channels NSFW setting.
+  """
+  @type nsfw :: boolean()
 
-  @typedoc "Id of the last message sent"
+  @typedoc """
+  The id of the last message sent in the channel.
+  """
   @type last_message_id :: Message.id() | nil
 
-  @typedoc "The bitrate of the voice channel"
+  @typedoc """
+  The bitate of the channel.
+  """
   @type bitrate :: integer()
 
-  @typedoc "The user limit of the voice channel"
+  @typedoc """
+  The users rate limit.
+
+  Amount of seconds a user has to wait before sending another message (0-21600); bots, as well as users with the permission manage_messages or manage_channel, are unaffected
+  """
+  @typedoc since: "0.5"
+  @type rate_limit_per_user :: String.t() | nil
+
+  @typedoc """
+  The user limit of a voice channel.
+  """
   @type user_limit :: integer()
 
-  @typedoc "The recipients of the DM"
+  @typedoc """
+  A list of users in a group channel.
+  """
   @type recipients :: [User.t()]
 
-  @typedoc "The icon hash of the channel"
+  @typedoc """
+  The hash of the channels icon.
+  """
   @type icon :: String.t() | nil
 
-  @typedoc "The id of the DM creator"
+  @typedoc """
+  The id of the channels creator.
+  """
   @type owner_id :: User.id()
 
-  @typedoc "The application id of the group DM creator if it is bot-created"
+  @typedoc """
+  The id of the application that created the channel.
+  """
   @type application_id :: Snowflake.t() | nil
 
-  @typedoc "The id of the parent category for a channel"
+  @typedoc """
+  The id of the category channel that this channel is located under.
+  """
   @type parent_id :: Channel.id() | nil
 
-  @typedoc "When the last pinned message was pinned"
+  @typedoc """
+  Timestamp for the last pinned message.
```

Comment:
If we are updating this docstring, can you add a mention that this is ISO8601 format and `DateTime.from_iso8601/2` can be used to parse it if wanted?